cfme/tests/middleware/dballocator_methods.py

```diff
@@ -0,0 +1,96 @@
+ORACLE_12C_RAC = 'oracle12cRAC'
+SEP = '='
+
+
+def db_allocate(db_type, expiry=1):
+    """
+    Method for allocating database via DBAllocator tool by providing database type,
+    expiry time in minutes and the name of requestee.
+
+    It returns the result of DBAllocator allocate action in dictionary format,
+    which contains the connection properties to allocated database.
+    From returned properties, very important one is 'uuid',
+        which will be used for manually deallocating database.
+
+    Args:
+        db_type: type of the database which is registered in DBAllocator,
+            all possible types are in this class.
+        expiry: reservation duration in minutes after
+            which database will be automatically deallocated (optional).
+
+    Returns:
+        database connection properties as dictionary containing information:
+            db.password: password
+            hibernate.connection.password: password
+            hibernate41.dialect: org.hibernate.dialect.PostgresPlusDialect
+            broken: false
+            db.username: username
+            server_geo: location
+            dballoc.db_type: standard
+            db.jdbc_url: formatted jdbc url to connect
+            db.name: database name
+            db.hostname: database hostname
+            datasource.class.xa: XADataSource class
+            server_uid: server uid
+            hibernate33.dialect: org.hibernate.dialect{}Dialect
+            server_labels: server_label
+            hibernate.connection.username: username
+            db.jdbc_class: JDBC Driver Class
+            db.schema: schema name
+            hibernate.connection.driver_class: Driver Class
+            uuid=2f7afb52-d4ca-4c33-9017-b7f12bfc4c70
+            db.primary_label: label
+            db.port: port
+            server_label_primary: server label
+            hibernate.dialect: org.hibernate.dialect.{}Dialect
+            hibernate.connection.url: formatted jdbc url to connect
+            hibernate.connection.schema: schema name
+
+    Usage:
+        properties = db_allocate('oracle12c')
+    """
+    try:
+        db_allocate_url = conf.cfme_data.get(
+            'resources', {})['databases']['db_allocator']['db_allocate_url']
+        with closing(urlopen(db_allocate_url.format(db_type, expiry, REQUESTEE))) as http:
+            properties = http.read().strip()
+            return parse_properties(properties, SEP)
+    except HTTPError as e:
+        pytest.fail('Error {} while allocating database {}'.format(e, db_type))
+
+
+def db_deallocate(uuid):
+    """
+    Method for deallocating reserved database from DBAllocator tool by provided uuid.
+    uuid parameter was returned as a property during allocation.
+
+    Args:
+        uuid: uuid of database instance in DBAllocator tool.
+
+    Usage:
+        properties = db_allocate('oracle12c')
+        db_deallocate(properties['uuid'])
+    """
+    try:
+        db_deallocate_url = conf.cfme_data.get(
```

Comment:
this line will fail on missing data and should not be in the try/except for a different error